File: dataModuleTester/src/main/java/com/aneonex/bitcoinchecker/tester/volley/generic/GenericCheckerVolleyRequest.java
package com.aneonex.bitcoinchecker.tester.volley.generic;

import com.android.volley.Response.ErrorListener;
import com.android.volley.Response.Listener;
import com.aneonex.bitcoinchecker.datamodule.model.CheckerInfo;

public abstract class GenericCheckerVolleyRequest<T> extends GzipVolleyRequest<T> {
	
	protected final CheckerInfo checkerInfo;

	public GenericCheckerVolleyRequest(String url, CheckerInfo checkerInfo, Listener<T> listener, ErrorListener errorListener) {
		super(url, listener, errorListener);
		
		this.checkerInfo = checkerInfo;
	}
}

File: dataModuleTester/src/main/java/com/aneonex/bitcoinchecker/tester/volley/generic/GenericCheckerVolleyRequest.kt
package com.aneonex.bitcoinchecker.tester.volley.generic

import com.android.volley.Response
import com.aneonex.bitcoinchecker.datamodule.model.CheckerInfo
import com.aneonex.bitcoinchecker.datamodule.model.PostRequestInfo

abstract class GenericCheckerVolleyRequest<T>(url: String?, postRequestInfo: PostRequestInfo?, protected val checkerInfo: CheckerInfo, listener: Response.Listener<T>, errorListener: Response.ErrorListener)
    : GzipVolleyRequest<T>(url, postRequestInfo, listener, errorListener)